junit/src/main/java/org/kaazing/k3po/junit/rules/ScriptRunner.java

```diff
@@ -214,6 +214,7 @@ public void notifyBarrier(final String barrierName) throws InterruptedException
 
 
     public void dispose() throws Exception {
+        latch.awaitFinished();
```

Comment:
There are many latches and we can block on any of them (prepare, start, finished, barrier latches). Especially while debugging we can encounter some problems. If we want to avoid the k3po blocking we will have to rework the entire junit side with timeouts. From my point of view, this could be an entire refactoring work by itself.

As for the different exception handling, there is a difference between AssumptionViolatedException (that is normal and expected in a test that is failing and that should be passed further) and all the other exceptions that should be handled in k3po (or at least doing a cleanup). I don't think we can have a unified handling of the exceptions. Even in your second code block, the wait is on the result of the scriptFuture, not on a latch.
You can open an improvement issue if you really want to investigate further on this direction.